mobile/cmp/multiZoneGrid/MultiZoneGridModel.ts

```diff
@@ -0,0 +1,421 @@
+        const ret = this.gridModel.findColumn(this.gridModel.columns, field);
+        return ret.displayName;
+    }
+
+    //-----------------------
+    // Implementation
+    //-----------------------
+    private createGridModel(config: GridConfig): GridModel {
+        return new GridModel({
+            ...config,
+            sizingMode: 'standard',
+            cellBorders: true,
+            rowBorders: true,
+            stripeRows: false,
+            autosizeOptions: {mode: 'disabled'},
+            columns: this.getColumns()
+        });
+    }
+
+    private getColumns(): ColumnSpec[] {
+        return [
+            this.buildMultiZoneColumn(true),
+            this.buildMultiZoneColumn(false),
+            // Ensure all available columns are provided as hidden columns for lookup by multifield renderer
+            ...this.availableColumns
+        ];
+    }
+
+    private buildMultiZoneColumn(isLeft: boolean): ColumnSpec {
+        const topMappings = this.mappings[isLeft ? 'tl' : 'tr'],
+            bottomMappings = this.mappings[isLeft ? 'bl' : 'br'];
+
+        if (isEmpty(topMappings)) {
+            throw XH.exception(
+                `${isLeft ? 'Left' : 'Right'} column requires at least one top mapping`
+            );
+        }
+
+        // Extract the primary column from the top mappings
+        const primaryCol = new Column(this.findColumnSpec(topMappings[0]), this.gridModel);
+
+        // Extract the sub-fields from the other mappings
+        const subFields = [];
+        topMappings.slice(1).forEach(it => {
+            subFields.push({colId: it.field, label: it.showLabel, position: 'top'});
+        });
+        bottomMappings.forEach(it => {
+            subFields.push({colId: it.field, label: it.showLabel, position: 'bottom'});
+        });
+
+        return {
+            // Controlled properties
+            colId: isLeft ? 'left_column' : 'right_column',
+            headerName: primaryCol.headerName,
+            field: primaryCol.field,
+            flex: isLeft ? 2 : 1,
+            renderer: multiFieldRenderer,
+            rowHeight: Grid['MULTIFIELD_ROW_HEIGHT'],
+            resizable: false,
+            movable: false,
+            hideable: false,
+            appData: {
+                multiFieldConfig: {
+                    mainRenderer: primaryCol.renderer,
+                    delimiter: this.delimiter,
+                    subFields
+                }
+            },
+
+            // Properties inherited from primary column
+            absSort: primaryCol.absSort,
+            sortingOrder: primaryCol.sortingOrder,
+            sortValue: primaryCol.sortValue,
+            sortToBottom: primaryCol.sortToBottom,
+            comparator: primaryCol.comparator,
+            sortable: primaryCol.sortable,
+            getValueFn: primaryCol.getValueFn,
+
+            // Optional overrides
+            ...(isLeft ? this.leftColumnSpec : this.rightColumnSpec)
```

Comment:
wondering if the `xxxColumnSpec` should come *before* the primaryCol settings. Could be a more useful default like that, and the columns are also settable by end user...

Reply:
The main intention here is to allow the user to set their own column flex, which was managed with dedicated properties over in the Sencha version. I was trying to be flexible here, but maybe its better to lock it down to just offering dedicated flex props?